src/main/java/com/github/steveice10/mc/protocol/packet/ingame/clientbound/ClientboundAwardStatsPacket.java

```diff
@@ -27,44 +23,23 @@ public ClientboundAwardStatsPacket(ByteBuf in, MinecraftCodecHelper helper) {
         for (int index = 0; index < length; index++) {
             StatisticCategory category = helper.readStatisticCategory(in);
             int statisticId = helper.readVarInt(in);
-            Statistic statistic;
-            switch (category) {
-                case BREAK_BLOCK:
-                    statistic = new BreakBlockStatistic(statisticId);
-                    break;
-                case CRAFT_ITEM:
-                    statistic = new CraftItemStatistic(statisticId);
-                    break;
-                case USE_ITEM:
-                    statistic = new UseItemStatistic(statisticId);
-                    break;
-                case BREAK_ITEM:
-                    statistic = new BreakItemStatistic(statisticId);
-                    break;
-                case PICKED_UP_ITEM:
-                    statistic = new PickupItemStatistic(statisticId);
-                    break;
-                case DROP_ITEM:
-                    statistic = new DropItemStatistic(statisticId);
-                    break;
-                case KILL_ENTITY:
-                    statistic = new KillEntityStatistic(EntityType.from(statisticId));
-                    break;
-                case KILLED_BY_ENTITY:
-                    statistic = new KilledByEntityStatistic(EntityType.from(statisticId));
-                    break;
-                case CUSTOM:
-                    statistic = CustomStatistic.from(statisticId);
-                    break;
-                default:
-                    throw new IllegalStateException();
-            }
+            Statistic statistic = switch (category) {
```

Comment:
let's keep the default case too?

Reply:
This code will not compile if not all enum values are implemented.

Reply:
Sure, but only if all enums are implemented... I'd rather have a default throw that's not needed most of the time that can catch issues/missed updates/unsupported modded scenarios 
It's what we do with Geyser atleast, I don't see why it wouldn't fit here

Reply:
Enums are not to be extended at runtime.

Reply:
That's not what I'm concerned about, I'm aware of that :)
A default case could e.g. warn about a missed new enum that may not be implemented in MCPL when there's a Minecraft update.

Reply:
It doesn't, it just throws a exception at runtime. In this case the code does not compile when you missed handling a new enum option.

Reply:
that still assumes the enum was added to mcpl :p